examples/text_to_image/train_text_to_image_sdxl.py

```diff
@@ -1013,9 +992,17 @@ def compute_time_ids(original_size, crops_coords_top_left):
                     # Since we predict the noise instead of x_0, the original formulation is slightly changed.
                     # This is discussed in Section 4.2 of the same paper.
                     snr = compute_snr(timesteps)
-                    mse_loss_weights = (
+                    base_weight = (
                         torch.stack([snr, args.snr_gamma * torch.ones_like(timesteps)], dim=1).min(dim=1)[0] / snr
                     )
+
+                    if noise_scheduler.config.prediction_type == "v_prediction":
+                        # Velocity objective needs to be floored to an SNR weight of one.
+                        mse_loss_weights = base_weight + 1
```

Comment:
Why don't we do:
```py
mse_loss_weights = max(base_weight, 1) 
```

if we want to "floor" the SNR weight?

Reply:
`max` would be `ceil`, here we're just adding one to ensure a value of zero is a minimum of 1. we don't want to blindly overwrite minimal values with 1's, we want to scale all of them.

Reply:
`mse_loss_weights = max(base_weight, 1)` in that case works more like floor - e.g. `mse_loos_weights` is equal to at least 1 or higher. 

Can you explain in a bit more detail why we want to add 1 here all the time?

Reply:
certainly. when using v-prediction, there is an infinite SNR at the final timestep. if you do not floor the weight of this to 1, it will lead to a NaN loss value and the model fails to produce useable predictions.

this was observed in the original min-SNR paper, and has since been implemented in the reference code upstream.